File: examples/greeting-service/src/test/java/greeting/example/GreetingControllerTest.java
package greeting.example;

import io.micronaut.test.annotation.MicronautTest;
import org.junit.jupiter.api.Test;

import javax.inject.Inject;
import static org.junit.jupiter.api.Assertions.*;

@MicronautTest
class GreetingControllerTest {

    @Inject
    GreetingClient greetingClient;

    @Test
    void testGreetingService() {
        assertEquals(
                "Hola, John!",
                greetingClient.greet("John").getContent()
        );
    }
}

File: examples/greeting-service-kotlin/src/test/kotlin/greeting/example/GreetingControllerTest.kt
package greeting.example

import io.micronaut.test.annotation.MicronautTest
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

@MicronautTest
class GreetingControllerTest {

    @Inject
    lateinit var greetingClient : GreetingClient

    @Test
    fun testGreetingService() {
        assertEquals(
                "Hola, John!",
                greetingClient.greet("John").content
        )
    }
}
